net/graphsync_fetcher.go

```diff
@@ -319,10 +331,25 @@ type requestPeerFinder struct {
 	triedPeers  map[peer.ID]struct{}
 }
 
-func newRequestPeerFinder(peerTracker graphsyncFallbackPeerTracker, initialPeer peer.ID) *requestPeerFinder {
-	pri := &requestPeerFinder{peerTracker, initialPeer, make(map[peer.ID]struct{})}
-	pri.triedPeers[initialPeer] = struct{}{}
-	return pri
+func newRequestPeerFinder(peerTracker graphsyncFallbackPeerTracker, initialPeer peer.ID) (*requestPeerFinder, error) {
+	pri := &requestPeerFinder{
+		peerTracker: peerTracker,
+		triedPeers:  make(map[peer.ID]struct{}),
+	}
+
+	// Use the initialPeer's ID if it is our own
+	if peerTracker.Self() == initialPeer {
```

Comment:
Please leave some more comments here, explicitly calling out that `initialPeer` is ignored if not self (and why). It should probably be renamed to be descriptive rather than prescriptive - maybe `originatingPeer`? Is that information received over libp2p pubsub even verifiable? Do we require signed messages?

If all we need is a boolean here, it would be even simpler to just pass a boolean.

Reply:
I like boolean and will do.